src/exe/colmap.cc

```diff
@@ -1256,6 +1256,146 @@ int RunModelAnalyzer(int argc, char** argv) {
   return EXIT_SUCCESS;
 }
 
+void WriteComparisonErrors(const std::string& path,
+                           const std::vector<double>& rotation_errors,
+                           const std::vector<double>& translation_errors,
+                           const std::vector<double>& proj_center_errors) {
+  std::ofstream file(path, std::ios::trunc);
+  CHECK(file.is_open()) << path;
+
+  file.precision(17);
+  file << "# Model comparison pose errors; one entry per common image"
+       << std::endl;
+  file << "# <rotation error (deg)>, <translation error>, <proj center error>"
+       << std::endl;
+  for (int i = 0; i < rotation_errors.size(); ++i) {
```

Comment:
size_t

Reply:
Could benefit from `CHECK_EQ(rotation_errors.size(), translation_errors.size());` and `CHECK_EQ(rotation_errors.size(), proj_center_errors.size());`